src/angular/projects/spark-core-angular/src/lib/components/sprk-link/sprk-link.component.ts

```diff
@@ -61,36 +61,63 @@ export class SparkLinkComponent implements OnInit {
   isExternal = false;
 
   ngOnInit() {
+    // Sets the default href if none provided
     if (this.href === '' || this.href === null || this.href === undefined) {
       this.href = '#';
+      return;
+    }
+    // Build jump link's href
+    if (this.isJumpLink(this.href)) {
+      this.href = `${this.getPathWithoutHash(this.router.url)}${this.href}`;
     }
   }
 
   isExternalLink(value): boolean {
     return new RegExp('^.*:', 'i').test(value);
   }
 
+  isJumpLinkWithPage(value): boolean {
+    return new RegExp('^.*#.+', 'i').test(value);
+  }
+
   isJumpLink(value): boolean {
     return new RegExp('^#.+', 'i').test(value);
   }
 
-  isDefault(value): boolean {
+  isNoActionLink(value): boolean {
     return value === '#';
   }
 
+  getPathWithoutHash(value): string {
+    return value.split('#')[0];
+  }
+
+  scrollToId() {
+    const elementID = this.href.split('#').pop();
+    const element: HTMLElement = document.getElementById(elementID);
```

Comment:
what about `<a name="something">`?

http://www.tagindex.net/html/link/a_name.html

Reply:
Name is obsolete in HTML5